src/reference/asciidoc/kafka.adoc

```diff
@@ -810,7 +829,23 @@ containerProps.setMessageListener(new MessageListener<Integer, String>() {
     ...
 });
 DefaultKafkaConsumerFactory<Integer, String> cf =
-                        new DefaultKafkaConsumerFactory<Integer, String>(consumerProps());
+                        new DefaultKafkaConsumerFactory<>(consumerProps());
+KafkaMessageListenerContainer<Integer, String> container =
+                        new KafkaMessageListenerContainer<>(cf, containerProps);
+return container;
+----
+====
+
+Note that when creating a `DefaultKafkaConsumerFactory`, using the constructor that just takes in the properties as above means that key and value `Deserializer` classes are picked up from configuration.
+Alternatively, `Deserializer` instances may be passed to the `DefaultKafkaConsumerFactory` constructor for key and/or value, in which case all Consumers share the same instances.
+Another option is to provide Suppliers of Deserializers that will be used to obtain separate `Deserializer` instances for each `Consumer`:
```

Comment:
If you'd use `Supplier<Deserializer>` here, you wouldn't miss that it is not a code.